app/models/mixins/scanning_mixin.rb

```diff
@@ -0,0 +1,377 @@
+    # self.last_sync_on = Time.new.utc  if updated == true
+    self.last_sync_on = Time.at(xml_node.root.attributes["created_on"].to_i).utc if updated == true && xml_node.root.attributes["created_on"]
+    save
+    hardware.save unless hardware.nil?
+  end
+
+  def scan_queue(userid = "system", options = {})
+    MiqQueue.put(
+      :class_name  => self.class.base_class.name,
+      :instance_id => id,
+      :method_name => "scan",
+      :args        => [userid, options]
+    )
+  end
+
+  # Call the miqhost webservice to do the SyncMetadata operation
+  def sync_metadata(category, options = {})
+    _log.debug "category=[#{category}] [#{category.class}]"
+    options = {
+      "category"    => category.join(","),
+      "from_time"   => last_drift_state_timestamp.try(:to_i),
+      "taskid"      => nil,
+      "target_id"   => id,
+      "target_type" => self.class.base_class.name
+    }.merge(options)
+    host = options.delete("host")
+    options = {
+      "args"        => [path],
+      "method_name" => "sync_metadata",
+      "vm_guid"     => guid # TODO: target_guid
+    }.merge(options)
+    ost = OpenStruct.new(options)
+    host.queue_call(ost)
+  rescue => err
+    _log.log_backtrace(err)
+  end
+
+  # Call the miqhost webservice to do the ScanMetadata operation
+  def scan_metadata(category, options = {})
+    _log.info "category=[#{category}] [#{category.class}]"
+    options = {
+      "category"    => category.join(","),
+      "taskid"      => nil,
+      "target_id"   => id,
+      "target_type" => self.class.base_class.name
+    }.merge(options)
+    host = options.delete("host")
+    # If the options hash has an "args" element, remove it and add it to the "args" element with self.path
+    miqhost_args = Array(options.delete("args"))
+    options = {
+      "args"        => [path] + miqhost_args,
+      "method_name" => "scan_metadata",
+      "vm_guid"     => guid # TODO: target_guid
+    }.merge(options)
+    ost = OpenStruct.new(options)
+    host.queue_call(ost)
+  rescue => err
+    _log.log_backtrace(err)
+  end
+
+  def scan_profile_list
+    ScanItem.get_default_profiles
+  end
+
+  def scan_profile_categories(scan_profiles)
+    cat_scan_list = []
+    begin
+      # Loop over all profiles and add category scan items to the array
+      scan_profiles.to_miq_a.each do |p|
+        p["definition"].each do |d|
+          if d["item_type"] == "category"
+            d["definition"]["content"].to_miq_a.each { |item| cat_scan_list << item["target"] }
+          else
+            cat_scan_list |= ["profiles"]
+          end
+        end
+      end
+    rescue
+    end
+
+    # If we get through the profile check and do not have anything pass the default list
+
+    cat_scan_list.uniq!
+    cat_scan_list = self.class.default_scan_categories_no_profile if cat_scan_list.empty?
+    cat_scan_list
+  end
+
+  def scan_on_registered_host_only?
+    false
+  end
+
+  def scan_via_miq_vm(miqVm, ost)
```

Comment:
@roliveri can you rename this to something more generic? (Consider I'll have to call it from containers as well)